compiler/noirc_frontend/src/tests.rs

```diff
@@ -3456,6 +3456,9 @@ fn arithmetic_generics_rounding_fail_on_struct() {
 
 #[test]
 fn unconditional_recursion_fail() {
+    // These examples are self recursive top level functions, which actually
+    // would not be inlined now, but this error comes from the compilation checks,
```

Comment:
```suggestion
    // would not be inlined, but this error comes from the compilation checks,
```
nit: don't think we need to specify "now" as unless you check the git history someone reading this comment does not know what was before "now".

Reply:
Ok we can leave it then. It was really just a personal preference nit

Reply:
Rewrote it anyway. It was a note for me because I was scratching my head again why this doesn't fail when the other one does.